texts/issues/messages.yml

```diff
@@ -1,63 +1,444 @@
+    details can be found at [should we write a guide or link elsewhere?]
+  details: [currentUrl, sourceUrl]
+
 -
   type: ATTACHMENT_NOT_VALID
-  title: A document is not valid
+  severity: ERROR
+  trigger: |
+    We check that a download is a pdf by checking the first 5 bytes contain
+    %pdf- If this is false, we set the PDF_NOT_VALID. In most cases, the issue
+    is replaced by a more specific issue as this issue is always triggered
+    when downloading HTML files.
+  title: The downloaded file was not a valid validate
+  description: |
+    We were unable to read the
+    {currentUrl}
+  resolution: |
+    We currently only support downloading pdf files. If the file is a pdf, it
+    may not conform to the pdf standard. We suggest recreating the file or
+    opening it in Adobe Reader to validate the file.
+  details: [currentUrl]
+
 -
-  type: NO_FULL_TEXT_LINKS
-  title: No attachments found
+  type: NO_VALID_FULLTEXT_LINKS_EXTRACTED
+  matches: [NO_FULL_TEXT_LINKS]
+  severity: WARNING
+  trigger: |
+    After downloading an HTML page, we attempt to extract links. If 0 are
+    found, this issue is triggered.
+  title: No valid attachment links found
+  description: |
+    We downloaded the repository metadata page for this item, but we were
+    unable to find any full text links
+  resolution: |
+    Ensure that dc:identifier contains a direct link to the full text. We only
+    support the following file types: .pdf
+  details: [currentUrl]
+
 -
-  type: EXTERNAL_UNKNOWN
-  title: No full text was found due to unknown external reasons
+  hidden: true
+  type: NO_ATTACHMENT_FOUND_NO_REPORTED_FAILURES
+  matches: [NON_EXISTENT_PAGE_ATTACHMENT]
+  severity: WARNING
+  trigger: |
+    This issue is triggered when no specific issue was detected and reported.
+
+    If this issue is stored, it means we need to identify and find the root
+    cause.
+
+-
+  hidden: true
+  type: ATTACHMENT_PROCESSING_FAILURE
+  matches: []
+  severity: ERROR
+
 -
-  type: ENCRYPTED_ATTACHMENT
-  title: Full text has digital restrictions
+  type: ATTACHMENT_ENCRYPTED
+  matches: [ENCRYPTED_ATTACHMENT]
+  severity: ERROR or WARNING
+  trigger: |
+    This issue is triggered when a pdf has DRM enabled.
+
+    If we are able to bypass, a Warning is given
+
+    If we are not, an Error
+  title: The pdf is encrypted
+  description: |
+    The pdf has Document Security enabled. This digitally restricts how the
+    pdf is opened and used. For example, it may prevent editing, printing or
+    viewing the full text via a screen reader.
+
+    If this is message is issued with a Warning, we were able to bypass the
+    security settings
+
+    If this message is an error, we were unable to bypass the security and we
+    have not imported the fulltext.
+  resolution: |
+    We recommend that Document Security is removed from the PDF.
+    [https://helpx.adobe.com/uk/acrobat/using/securing-pdfs-passwords.html#remove_password_security]
```

Comment:
```suggestion
  trigger: |
    This issue is triggered when a pdf has DRM enabled.

    If we are able to bypass, a WARNING is given, if we are not, an ERROR.
  title: The full text is encrypted
  description: |
    The PDF file has Document Security enabled. This digitally restricts how the
    document is opened and used. For example, it may prevent editing, printing
    or viewing the full text via a screen reader.

    We recommend to remove Document Security from the full text document.
  resolution:
    -
      caption: Learn more
      url: https://helpx.adobe.com/uk/acrobat/using/securing-pdfs-passwords.html#remove_password_security
    -
      caption: Mark resolved
```

Reply:
You've changed the title from PDF to full text, but the description is very specific for pdfs 

If we do end up supporting other documents, we'll either need a way to report different messages for each file format, or each file format requires it's own issue status for encrypted content. Any thoughts on this?

Reply:
Re: learn more, it is possible to have multiple "learn more" links?

Reply:
It's possible to have as many links as we want. However, from the point of user experience, I prefer to have no more than 2.

I agree with the comment about the title and will change it back. Thank you!

Reply:
However, I am not so sure. I wanted to link the issue to the part of the metadata right from the title. We expose this part (PDF) as a 'Full text' property. I wanted to point out that the problem happened there.

Also, from the point of the system, I am not sure we will download and expose multiple documents in different formats even if we support multiple formats.